README.md

```diff
@@ -173,7 +173,8 @@ to add this to your `proguard-rules.pro`:
 
 ```proguard
 -keepattributes *Annotation*, InnerClasses
--dontnote kotlinx.serialization.SerializationKt
+-dontnote kotlinx.serialization.AnnotationsKt # core serialization annotations
+-keep,includedescriptorclasses class kotlinx.serialization.json.**$$serializer { *; } # JsonElement serializers
```

Comment:
Unless I misunderstood something these two lines did not work for me as I kept having this issue:
```
Non-fatal Exception: java.lang.NoClassDefFoundError
kotlinx.serialization.json.JsonObjectSerializer
```

What I'm using and it seems to work in production now is this:
```
-keepattributes *Annotation*, InnerClasses
-dontnote kotlinx.serialization.AnnotationsKt # core serialization annotations
-keep,includedescriptorclasses class kotlinx.serialization.json.** { *; }
-keep,includedescriptorclasses class <APP PACKAGE NAME>.**$$serializer { *; }
-keepclassmembers class <APP PACKAGE NAME>.** {
    *** Companion;
}
-keepclasseswithmembers class <APP PACKAGE NAME>.** {
    kotlinx.serialization.KSerializer serializer(...);
}
```

The line `-keep,includedescriptorclasses class kotlinx.serialization.json.** { *; }` did the fix for me but I understand it's probably too broad.

Reply:
The same here.

Reply:
Probably should be `-keep,includedescriptorclasses class kotlinx.serialization.json.**Serializer { *; }`

Reply:
I'll give it a go now and let you know. My fix was more of a panic fix given it ended up in production :sweat_smile:

Reply:
I'm still getting the java.lang.ExceptionInInitializerError
```
 Caused by: kotlinx.serialization.SerializationException: Serializer for class 'JsonElement' is not found.
Mark the class as @Serializable or provide the serializer explicitly.
    at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:104)
    at kotlinx.serialization.internal.PlatformKt.platformSpecificSerializerNotRegistered(Platform.kt:29)
    at kotlinx.serialization.SerializersKt__SerializersKt.serializer(Serializers.kt:56)
    at kotlinx.serialization.SerializersKt.serializer(Unknown Source:1)
    at kotlinx.serialization.SerializersKt__SerializersKt.builtinSerializerOrNull$SerializersKt__SerializersKt(Serializers.kt:76)
    at kotlinx.serialization.SerializersKt__SerializersKt.serializerByKTypeImpl$SerializersKt__SerializersKt(Serializers.kt:66)
    at kotlinx.serialization.SerializersKt__SerializersKt.serializer(Serializers.kt:38)
    at kotlinx.serialization.SerializersKt.serializer(Unknown Source:1)
    at kotlinx.serialization.json.JsonObjectSerializer$JsonObjectDescriptor.<init>(JsonElementSerializers.kt:21)
    at kotlinx.serialization.json.JsonObjectSerializer$JsonObjectDescriptor.<clinit>(JsonElementSerializers.kt:148)
```

This is the setup after the change:
```
-keepattributes *Annotation*, InnerClasses
-dontnote kotlinx.serialization.AnnotationsKt # core serialization annotations
-keep,includedescriptorclasses class kotlinx.serialization.json.**$$serializer { *; }
-keep,includedescriptorclasses class kotlinx.serialization.json.**Serializer { *; }
-keep,includedescriptorclasses class <PACKAGE_NAME>.**$$serializer { *; }
-keepclassmembers class <PACKAGE_NAME>.** {
    *** Companion;
}
-keepclasseswithmembers class <PACKAGE_NAME>.** {
    kotlinx.serialization.KSerializer serializer(...);
}
```